src/adapters/consumer/index.ts

```diff
@@ -206,64 +211,78 @@ export class Consumer {
     }
 
     jsonToCots(json: any) {
+        if (!json || !json.data) {
+            console.error("Invalid JSON: ", json)
+            return []
+        }
+
         const data = json.data
```

Comment:
it looks like data was undefined so I added in checks for both json and json.data jic. 
Also added in some additional error handling lower down, needs to be more but as a quick fix.